File: 初级算法/Math/FizzBuzz.js
/**
 * Fizz Buzz
 * source: https://leetcode-cn.com/explore/interview/card/top-interview-questions-easy/25/math/60/
 */

var fizzBuzz = function(n) {
    var stringArray = new Array();

    for (var i = 1; i <= n; i++) {
        if (i%3 == 0 && i%5==0) {
            stringArray.push('FizzBuzz');
        } else if (i%3 == 0) {
            stringArray.push('Fizz');
        } else if (i%5 == 0) {
            stringArray.push('Buzz');
        } else {
            stringArray.push(i.toString());
        }
    }

    return stringArray;
};



console.log(fizzBuzz(15));

File: 初级算法/Swift/Math/fizzBuzz.swift
/**
 * Fizz Buzz
 * source: https://leetcode-cn.com/explore/interview/card/top-interview-questions-easy/25/math/60/
 */

import Foundation

class Solution {
    func fizzBuzz(_ n: Int) -> [String] {
        guard n >= 1 else {
            return []
        }
        
        var strings: [String] = []
        for i in 1...n {
            var nString = "\(i)"
            if i%3 == 0 && i%5 == 0 {
                nString = "FizzBuzz"
            } else if i%3 == 0 {
                nString = "Fizz"
            } else if i%5 == 0 {
                nString = "Buzz"
            }
            
            strings.append(nString)
        }
        
        return strings
    }
}


let n = 10
print(Solution().fizzBuzz(n))


